### PR TITLE
Update recipe for Segtools 1.2.1 for both Python 2 and Python 3

### PR DESCRIPTION
**Pull request by EricR86**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [ ] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by EricR86** [comment]
I cannot figure out what the version conflict is coming from. The conflict error message is very confusing:
```python
conda.exceptions.UnsatisfiableError: The following specifications were found to be in conflict:
  - segtools==1.2.1=py35r341h39e3cac_0
```

I can confirm I can reproduce this on conda-build locally and I can also `python setup.py install` segtools on a fresh Python 3.5 (or 3.6) install. I just can't figure out what is causing the conflict from the message given.

**Comment by EricR86** [comment]
I've recently updated my conda and "conda build" and I can no longer reproduce the error reported on the build machine locally. Is there anyway to exactly reproduce the error reported on the build machine or show what version of conda/conda build it is using?

**Comment by EricR86** [comment]
It seems this problem was fixed due to an odd issue with preserve egg directories being unnecessarily added to the recipe. If there's anything else necessary for a merge please let me know.

File: recipes/segtools/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,13 @@
...
-    - segtools-flatten = segtools.flatten:main
-    - segtools-gmtk-parameters = segtools.gmtk_parameters:main
-    - segtools-html-report = segtools.html:main
-    - segtools-length-distribution = segtools.length_distribution:main
-    - segtools-nucleotide-frequency = segtools.nucleotide_frequency:main
-    - segtools-overlap = segtools.overlap:main
-    - segtools-preprocess = segtools.preprocess:main
-    - segtools-relabel = segtools.relabel:main
-    - segtools-signal-distribution = segtools.signal_distribution:main
-    - segtools-transition = segtools.transition:main
```
Comment (daler) [inline recipes/segtools/meta.yaml]: Just checking: this package now supports py3 and no longer need entry points?

File: recipes/segtools/meta.yaml [hunk 2]
```diff
@@ -45,7 +28,8 @@ requirements:
     - r-reshape2
     - r-cluster
     - gmtk
-    - genomedata
+    - textinput
+    - genomedata # [py2k]
```
Comment (daler) [inline recipes/segtools/meta.yaml]: genomedata not needed when using py3?

**Comment by daler** [comment]
Never mind on the py3 part . . . while main part of docs at https://hoffmanlab.org/proj/segtools/ says 2.7, the matrix at https://hoffmanlab.org/proj/segtools/doc/1.2.1/segtools.html#python-2-3-compatibility-status has tool-specific compatibilities.

**Pull request merged**